csharp/src/Microsoft.ML.OnnxRuntime/OrtAllocator.shared.cs

```diff
@@ -310,6 +311,69 @@ protected override bool ReleaseHandle()
         #endregion
     }
 
+    /// <summary>
+    /// This class represents an arbitrary buffer of memory
+    /// allocated and owned by the user. It can be either a CPU, GPU or other device memory
+    /// that can be suitably represented by IntPtr.
+    /// This is just a composite of the buffer related information.
+    /// The memory is assumed to be pinned if necessary and usable immediately
+    /// in the native code.
+    /// </summary>
+    public class OrtExternalAllocation
+    {
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="memInfo">use to accurately describe a piece of memory that this is wrapping</param>
+        /// <param name="shape">shape of this buffer</param>
+        /// <param name="elementType">element type</param>
+        /// <param name="pointer">the actual pointer to memory</param>
+        /// <param name="sizeInBytes">size of the allocation in bytes</param>
+        public OrtExternalAllocation(OrtMemoryInfo memInfo, long[] shape, Tensors.TensorElementType elementType, IntPtr pointer, long sizeInBytes)
+        {
+            Type type;
+            int width;
+            TensorElementTypeConverter.GetTypeAndWidth(elementType, out type, out width);
+            if (width < 1)
+            {
```

Comment:
nit: can we update GetTypeAndWidth to return true/false if successful? if not, would checking type==null be a little more obvious than width < 1?